core/domain/classifier_services.py

```diff
@@ -100,8 +101,39 @@ def classify_string_classifier_rule(state, normalized_answer):
             training_examples.extend([
                 [doc, [str(answer_group_index)]]
                 for doc in classifier_rule_spec.inputs['training_data']])
-    if len(training_examples) > 0:
+
+    exploration = exp_services.get_exploration_by_id(exp_id)
+    state = exploration.states[state_name]
+    algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+        'TextInput']['classifier_id']
+    if state.classifier_model_id:
+        classifier = get_classifier_by_id(state.classifier_model_id)
+        sc_dict = classifier.cached_classifier_data
+        sc = classifier_registry.Registry.get_classifier_by_algorithm_id(
+            algorithm_id)
+        sc.from_dict(sc_dict)
```

Comment:
So, basically, the question comes down to whether we can find a way to not create an empty class ever and pass parameters to the constructor always. But, for this particular class, the constructor does not take any parameters. The only way to go about this right now is to initialize an empty class (which initializes the values of all the constants) and then modify it to add params. 

Long term though, if we need to implement this objective, we can implement a constructor which takes parametrized values such that we initialize all the params inside the constructor.

Also, I think combining the initialization and training into one process may be a really good idea. I believe this is implementable and that we can proceed with the issue. I can make an issue for this if we decide to go ahead with this.

On a side note, should I revert the latest commits regarding the @classmethod updates for from_dict()?